src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/RegionServiceImpl.java

```diff
@@ -18,13 +36,99 @@
+        if (regionId.strip().equalsIgnoreCase(ALL)) {
+            return retrieveAllRegions(false);
+        }
+
+        //otherwise generate list of IDs from regionId String
+        List<String> regionsIdList =
+            checkIfValidCsvIdentifiersAndReturnList(regionId, EXCEPTION_MSG_NO_VALID_REGION_ID_PASSED);
 
-        if (isEmpty(region)) {
-            throw new ResourceNotFoundException("No Region found with the given description: " + description);
+        //then check if List contains 'ALL' and the ID for National
+        if (isListContainsTextIgnoreCase(regionsIdList, ALL) && regionsIdList.contains("1")) {
+            return retrieveAllRegions(true);
+        } else if (isListContainsTextIgnoreCase(regionsIdList, ALL)) {
+            return retrieveAllRegions(false);
         }
 
-        return new LrdRegionResponse(region);
+        //remove invalid IDs from the list
+        checkForInvalidIdentifiersAndRemoveFromIdList(
+            regionsIdList, ALPHA_NUMERIC_REGEX, log, loggingComponentName, EXCEPTION_MSG_NO_VALID_REGION_ID_PASSED);
+
+        List<Region> regions = regionRepository.findByRegionIdIn(regionsIdList);
+
+        if (isEmpty(regions)) {
+            throw new ResourceNotFoundException("No Region(s) found with the given Region ID: " + regionsIdList);
+        }
+
+        return regions.stream().map(LrdRegionResponse::new).collect(Collectors.toList());
+    }
+
+    public List<LrdRegionResponse> retrieveRegionByRegionDescription(String description) {
+        checkStringContainsMoreThanOneConsecutiveComma(description);
+
+        //generate list of Descriptions from description String
+        List<String> regionDescriptionsList =
+            checkIfValidCsvIdentifiersAndReturnList(description, EXCEPTION_MSG_NO_VALID_REGION_DESCRIPTION_PASSED);
+
+        //remove invalid Regions from the list
+        checkForInvalidIdentifiersAndRemoveFromIdList(
+            regionDescriptionsList, REGION_NAME_REGEX, log,
+            loggingComponentName, EXCEPTION_MSG_NO_VALID_REGION_DESCRIPTION_PASSED
+        );
+
+        List<Region> regions = regionRepository.findByDescriptionInIgnoreCase(regionDescriptionsList);
+
+        if (isEmpty(regions)) {
+            throw new ResourceNotFoundException("No Region(s) found with the given Region Description(s): "
+                                                    + regionDescriptionsList);
+        }
+
+        return regions.stream().map(LrdRegionResponse::new).collect(Collectors.toList());
+    }
+
+    public List<LrdRegionResponse> retrieveAllRegions(boolean isNationalRequired) {
+        List<Region> regions;
+
+        if (isNationalRequired) {
+            regions = regionRepository.findAll();
+        } else {
+            regions = regionRepository.findByDescriptionNotIgnoreCase("National");
+        }
+
+        return regions.stream().map(LrdRegionResponse::new).collect(Collectors.toList());
+    }
+
+    public void isRegionIdParamValid(String param) {
```

Comment:
Is this supposed to be a public method? It believe it should be a private method.

Reply:
Addressed here: https://github.com/hmcts/rd-location-ref-api/pull/273/commits/d01485f4e3bdc6df6114f6e67e79d7b37497f775